website/content/en/docs/building-operators/golang/project_migration_guide.md

```diff
@@ -50,133 +66,185 @@ multigroup: true
+And then, you can use `log` instead of `reqLogger`. Example:
 
-Refer to [`memcached_controller.go`][kb_memcached_controller] for the example implementation of `Reconcile()` logic which would be present in `api/v1alpha1/memcached_controller.go` of Memcached operator.
+```go
+log.Info("Memcached resource not found. Ignoring since object must be deleted")
+```
 
-## Generate CRDs
+- The `r.client.Action` will be replaced with `r.Action`. (e.g replace `r.client.Get` with `r.Get`)
+- The `r.scheme` will be replaced with `r.Scheme`
 
-Run [`make manifests`][generate_crd] to generate CRD manifests. They would be generated inside the `config/crd/bases` folder.
+### Setting the RBAC permissions
 
-## Configuring your test environment
+The RBAC permissions now are configured via markers whcih will generate and update the manifest files present in `config/rbac/`. See that on top of the `Reconcile()` method you will find them. For our Memcache example it will be: 
 
-Projects are scaffolded with unit tests that utilize the [envtest](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/envtest)
-library, which requires certain Kubernetes server binaries be present locally.
-Installation instructions can be found [here][env-test-setup].
+```go
+// +kubebuilder:rbac:groups=cache.example.com,resources=memcacheds,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=cache.example.com,resources=memcacheds/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=apps,resources=services,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=pods,verbs=get;list
+// +kubebuilder:rbac:groups=core,resources=services,verbs=get;list;watch;create;update;patch;delete
+```
 
-## Operator Manifests
+However, you might configure additional permissions for your project then, see [RBAC markers][rbac_markers] doc for further information. 
 
-### Operator deployment manifests
+**Note** You might would like to check the example [[memcached_controller.go]][memcached_controller]. 
 
-The operator deployment manifest [`deploy/operator.yaml`][deployment_yaml] from the old project should be copied over to `config/manager/manager.yaml` in the new project.
+## Migrating the Main.go
 
-**Note**:
-The kustomize file requires the operator deployment manifest to have the field `namespace` which is missing in the `deploy/operator.yaml` manifest of SDK project.
-> **// TODO:** Explain label propagation for metrics collection.
+If you are using third-party API's in your project for example then, it means that you need to copy your customizations from `cmd/manager/main.go` to `cmd/main.go`. You’ll also need to ensure all of the needed schemes have been registered.
 
```

Comment:
I think a few more things can be mentioned for the main.go/manager:
* addition of health-probes needs to be added by defining the port it should listen to, preferably it should be done by scaffolding code
* Setting `Namespace` in the controller.Opts similar to WATCH_NAMESPACE for operator-sdk generated code.

Reply:
Hi @davidkarlsen, 

I added more info. Regards `HealthProbeBindAddress` it is not customization that we described for the legacy layout. So, IMO we need to try also be less verbose as possible and more generic since we will not able to cover all possibilities. However, shows quite intuitive as well that both are the same fie.  

> preferably it should be done by scaffolding code
It shows a great idea wdyt about raise an issue for the repo for we do that? 

> Setting Namespace in the controller.Opts similar to WATCH_NAMESPACE for operator-sdk generated code.
It is described in the operator scope doc. However, I linked it and added more info about that as well as described that these env vars are no longer supported.

Reply:
the legacy layout provided a metrics endpoint, and it had an ´/healthz´ endpoint at port 8686. The new layout misses this ootb.